tests/upload.js

```diff
@@ -373,4 +373,172 @@ describe("File upload", function () {
+
+      const scope = nock("https://upload.imagekit.io/api")
+        .post("/v1/files/upload")
+        .basicAuth({ user: initializationParams.privateKey, pass: "" })
+        .reply(200, function (uri, requestBody) {
+          expect(this.req.headers["content-type"]).include("multipart/form-data; boundary=---------------------");
+          var boundary = this.req.headers["content-type"].replace("multipart/form-data; boundary=", "");
+          checkFormData({ requestBody, boundary, fieldName: "fileName", fieldValue: fileOptions.fileName });
+          checkFormData({ requestBody, boundary, fieldName: "file", fieldValue: fileOptions.file });
+          checkFormData({ requestBody, boundary, fieldName: "transformation", fieldValue: transformation });
+          done();
+        });
+
+      imagekit.upload(fileOptions, callback);
+    });
+    it("Should return error for an invalid transformation", async function () {
+      const fileOptions = {
+        fileName: "test_file_name",
+        file: "test_file",
+        responseFields: "tags, customCoordinates, isPrivateFile, metadata",
+        useUniqueFileName: false,
+        transformation: {},
+      };
+      var callback = sinon.spy();
+
+      imagekit.upload(fileOptions, callback);
+
+      var errRes = {
+        help: "",
+        message: "Invalid transformation parameter.",
+      };
+      expect(callback.calledOnce).to.be.true;
+      sinon.assert.calledWith(callback, errRes, null);
+    });
+    it("Should return error for an invalid pre transformation", async function () {
+      const fileOptions = {
+        fileName: "test_file_name",
+        file: "test_file",
+        responseFields: "tags, customCoordinates, isPrivateFile, metadata",
+        useUniqueFileName: false,
+        transformation: { pre: "" },
+      };
+      var callback = sinon.spy();
+
+      imagekit.upload(fileOptions, callback);
+
+      var errRes = {
+        help: "",
+        message: "Invalid pre transformation parameter.",
+      };
+      expect(callback.calledOnce).to.be.true;
+      sinon.assert.calledWith(callback, errRes, null);
+    });
+    it("Should return error for an invalid post transformation of type abs", async function () {
+      const fileOptions = {
+        fileName: "test_file_name",
+        file: "test_file",
+        responseFields: "tags, customCoordinates, isPrivateFile, metadata",
+        useUniqueFileName: false,
+        transformation: { post: [{ type: "abs", value: "" }] },
+      };
+      var callback = sinon.spy();
+
+      imagekit.upload(fileOptions, callback);
+
+      var errRes = {
+        help: "",
+        message: "Invalid post transformation parameter.",
+      };
+      expect(callback.calledOnce).to.be.true;
+      sinon.assert.calledWith(callback, errRes, null);
+    });
+    it("Should return error for an invalid post transformation of type transformation", async function () {
```

Comment:
New line missing. Same feedback for all PRs.

Reply:
done